src/DefaultBuilder/src/StaticWebAssets/StaticWebAssetsHelper.cs

```diff
@@ -0,0 +1,137 @@
+                    UseStaticWebAssetsCore(environment, manifest);
+                }
+            }
+        }
+
+        internal static void UseStaticWebAssetsCore(this IWebHostEnvironment environment, Stream manifest)
+        {
+            var staticWebAssetsFileProvider = new List<IFileProvider>();
+            var webRootFileProvider = environment.WebRootFileProvider;
+
+            var staticWebAssetContentRoots = Parse(manifest)
+                .Select(cr => new StaticWebAssetsFileProvider(cr.BasePath, cr.Path))
+                .OfType<IFileProvider>() // Upcast so we can insert on the resulting list.
+                .ToList();
+
+            if (staticWebAssetContentRoots.Count == 0)
+            {
+                return;
+            }
+            else
+            {
+                staticWebAssetContentRoots.Insert(0, webRootFileProvider);
+                environment.WebRootFileProvider = new CompositeFileProvider(staticWebAssetContentRoots);
+            }
+        }
+
+        internal static Stream ResolveManifest(IWebHostEnvironment environment)
+        {
+            var assembly = Assembly.Load(environment.ApplicationName);
+            if (assembly.GetManifestResourceNames().Any(a => a == StaticWebAssetsManifestName))
+            {
+                return assembly.GetManifestResourceStream(StaticWebAssetsManifestName);
+            }
+            else
+            {
+                // Fallback to physical file as we plan to use a file on disk instead of the embedded resource.
+                var filePath = Path.Combine(Path.GetDirectoryName(new Uri(assembly.CodeBase).LocalPath), $"{environment.ApplicationName}.StaticWebAssets.xml");
+                if (File.Exists(filePath))
+                {
+                    return File.OpenRead(filePath);
+                }
+                else
+                {
+                    // A missing manifest might simply mean that the feature is not enabled, so we simply
+                    // return early. Misconfigurations will be uncommon given that the entire process is automated
+                    // at build time.
+                    return null;
+                }
+            }
+        }
+
+        internal static IEnumerable<ContentRootMapping> Parse(Stream manifest)
+        {
+            var document = XDocument.Load(manifest);
+            if (!string.Equals(document.Root.Name.LocalName, ManifestRootElementName, StringComparison.OrdinalIgnoreCase))
+            {
+                throw new InvalidOperationException($"Invalid manifest format. Manifest root must be '{ManifestRootElementName}'");
+            }
+
+            var version = document.Root.Attribute(VersionAttributeName);
+            if (version == null)
+            {
+                throw new InvalidOperationException($"Invalid manifest format. Manifest root element must contain a version '{VersionAttributeName}' attribute");
+            }
+
+            if (version.Value != "1.0")
```

Comment:
One thought I have ... re:versioning. I think it's good that you put a version in the file-format, but that isn't really a get out of jail free card. SDKs support **ALL** versions of .NET, so we're not really free to *just* make breaking changes to the file-format, it has to be conscious of the version of the runtime.

Reply:
I don't expect you to resolve this in this PR 😆 - just felt like mentioning it.

Reply:
If we evolve the SDK we will evolve this code in sync. I think the SDK would include logic to generate a version of the manifest that is compatible with the given runtime being targeted.